lib/core/plugins/manager.js

```diff
@@ -648,7 +648,7 @@ class PluginsManager {
         throw strategyError.get('unauthorized_removal', strategyName);
       }
 
-      delete this.strategies[strategyName];
+      this.strategies[strategyName] = undefined;
```

Comment:
:warning: this causes a subtle bug

If you register a strategy dynamically, then unregister it, this will set `this.strategies[strategyName]` to undefined. But if you try to register a strategy again with the same name, the following check will be truthy: https://github.com/kuzzleio/kuzzle/blob/master/lib/core/plugins/manager.js#L522

Because even if set to `undefined`, the object `this.strategies` has a property named with the strategy name. This will trigger an attempt to unregister the strategy, but because it is undefined, the manager will throw a `strategy_not_found` error.

Thus you can never again re-register a strategy with the same name after it has been unregistered.